src/software/proto/message_translation/ssl_message_translator.cpp

```diff
@@ -0,0 +1,305 @@
+#include "software/proto/message_translation/ssl_message_translator.h"
+
+#include "shared/constants.h"
+#include "software/logger/logger.h"
+
+std::optional<SSL_FieldLineSegment> findLineSegment(
+    const google::protobuf::RepeatedPtrField<SSL_FieldLineSegment>& line_segments,
+    const std::string& name)
+{
+    for (const auto& segment : line_segments)
+    {
+        if (segment.name() == name)
+        {
+            return segment;
+        }
+    }
+
+    return std::nullopt;
+}
+
+std::optional<SSL_FieldCircularArc> findCircularArc(
+    const google::protobuf::RepeatedPtrField<SSL_FieldCircularArc>& circular_arcs,
+    const std::string& name)
+{
+    for (const auto& arc : circular_arcs)
+    {
+        if (arc.name() == name)
+        {
+            return arc;
+        }
+    }
+
+    return std::nullopt;
+}
+
+std::unique_ptr<Vector2f> createVector2f(const Point& point)
+{
+    auto vector2f = std::make_unique<Vector2f>();
+    vector2f->set_x(static_cast<float>(point.x()));
+    vector2f->set_y(static_cast<float>(point.y()));
+    return std::move(vector2f);
+}
+
+std::unique_ptr<SSL_FieldLineSegment> createFieldLineSegment(
+    const Segment& segment, const float thickness, const std::string& name,
+    const SSL_FieldShapeType& shape_type)
+{
+    if (thickness < 0)
+    {
+        throw std::invalid_argument("SSL_FieldLineSegment thickness must be >= 0");
```

Comment:
Is there any difference between using `throw std::invalid_argument` against using `assert`? Wanted to ask to keep consistency.

Reply:
We basically prefer exceptions in c++ code because they are recoverable (exceptions can be caught, assertions can not). Exceptions don't exist in C so we generally only use asserts there.

Reply:
It's worth noting as well that `asserts` should not generally be used for handling errors unless something _really_ badly goes wrong, because they will totally stop your program (which we generally try to avoid) and when we compile with all optimisations turned on, the compiler actually removes them all. They can be thought of as a helpful tool for debugging (much easier to see that an assert failed then see the effects of some invalid calculation downstream).